### DISCUSSION TITLE
why clickhouse-jdbc can't support a connection pool like hikari?

### DISCUSSION THREAD

**Discussion by Young0516** [ClickHouse/ClickHouse#56127] [Q&A]
in order to improve the performance from client. I get this question.

**Comment by mshustov** [comment]
let's continue in https://github.com/ClickHouse/clickhouse-java/issues/1480

**Reply by Young0516** [reply]
thank you

**Comment by mzitnik** [comment]
In JDBC driver `.getConnection` is generating an under the hood a single connection to the DB the heavy lifting of the connection pool is done outside the diver, by using.  Hikari externally solves the concurrency issue.

```
HikariConfig configuration = new HikariConfig();
configuration.setMaximumPoolSize(1);
configuration.setJdbcUrl("jdbc:ch:{hostname}:{port}/default?insert_quorum=auto&server_time_zone&server_version=22.13.1.24495");
configuration.setUsername(username);
configuration.setPassword(password);
configuration.setValidationTimeout(6000000);
HikariDataSource ds = new HikariDataSource(configuration); 
```

Use `Connection conn = ds.getConnection();` to get connection and you get it from hikari pool 
I will provide a sample with full mvn example for using
